### PR TITLE
support admin kubeconfig from URL

### PR DESCRIPTION
**Pull request by akostadinov**

### CODE REVIEW

**Comment by akostadinov** [comment]
@xxia FYI BUSHSLICER_CONFIG can be:
```
environments:
  ocp4:
    admin_creds_spec: "https://example.com/job/Launch%20Environment%20Flexy/56402/artifact/workdir/cucushift-oc40-standard-slave-k26jp-0/install-dir/auth/kubeconfig"
```

For local runs it can be easier:
```
export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC="https://example.com/job/Launch%20Environment%20Flexy/56402/artifact/workdir/cucushift-oc40-standard-slave-k26jp-0/install-dir/auth/kubeconfig"
```

**Comment by xingxingxia** [comment]
Test passes, thank you :+1:

File: lib/admin_credentials.rb [hunk 1]
```diff
@@ -43,22 +61,32 @@ def get
...
+
+  class AutoKubeconfigCredentials < AdminCredentials
+    def get
+      case opts[:spec]
+      when nil
+        MasterOsAdminCredentials.new(env, **opts).get
+      when %r{://}
+        URLKubeconfigCredentials.new(env, **opts).get
+      else
+        raise "unknown gredentials specification: #{opts[:spec]}"
```
Comment (pruan-rht) [inline lib/admin_credentials.rb]: minor typo should be `credentials` I think...

**Comment by xingxingxia** [comment]
I added short usage of the url spec and :bastion role in the description stanza for Runner-v3 params HOSTS and BUSHSLICER_CONFIG respectively so auto builders can know.

File: lib/admin_credentials.rb (hunk 1 above)
Comment (wjiangjay) [reply lib/admin_credentials.rb]: typo from Dvorak keyboard layout?

Comment (akostadinov) [reply lib/admin_credentials.rb]: Not really, more likely just being tired when typing that.

Comment (wjiangjay) [reply lib/admin_credentials.rb]: I make this guess as in Dvorak keyboard layout, "g" is the neighbor of "c". And thanks for make this happen for admin tag cases

**Pull request merged**